modules/api/src/main/java/org/apache/ignite/table/criteria/Criteria.java

```diff
@@ -17,9 +17,226 @@
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;))
+     * </pre>
+     *
+     * @param columnName Column name.
+     * @param condition Target condition.
+     * @return The created expression instance.
+     */
+    static Expression columnValue(String columnName, Condition condition) {
+        Criteria[] oldElements = condition.getElements();
+        var newElements = new Criteria[oldElements.length + 1];
+
+        newElements[0] = new Column(columnName);
+        System.arraycopy(oldElements, 0, newElements, 1, oldElements.length);
+
+        return new Expression(condition.getOperator(), newElements);
+    }
+
+    /**
+     * Creates the negation of the predicate.
+     *
+     * <p>For example:
+     * <pre>
+     *     not(columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;)))
+     * </pre>
+     *
+     * @param expression Expression.
+     * @return The created negation of the expression.
+     */
+    static Expression not(Expression expression) {
+        return new Expression(Operator.NOT, expression);
+    }
+
+    /**
+     * Creates the {@code and} of the expressions.
+     *
+     * <p>For example:
+     * <pre>
+     *     and(columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;)), columnValue(&quot;quantity&quot;, lessThan(20)))
+     * </pre>
+     *
+     * @param expressions Expressions.
+     * @return The created {@code and} expression instance.
+     */
+    static Expression and(Expression... expressions) {
+        return new Expression(Operator.AND, expressions);
+    }
+
+    /**
+     * Creates the {@code or} of the expressions.
+     *
+     * <p>For example:
+     * <pre>
+     *     or(columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;)), columnValue(&quot;category&quot;, equalTo(&quot;games&quot;)))
+     * </pre>
+     *
+     * @param expressions Expressions.
+     * @return The created {@code or} expressions instance.
+     */
+    static Expression or(Expression... expressions) {
+        return new Expression(Operator.OR, expressions);
+    }
+
+    /**
+     * Creates a condition that test the examined object is equal to the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;))
+     * </pre>
+     *
+     * @param <T> Value type.
+     * @param value Target value.
+     */
+    static <T> Condition equalTo(Comparable<T> value) {
+        return new Condition(Operator.EQ, new Parameter<>(value));
+    }
+
+    /**
+     * Creates a condition that test the examined object is equal to the specified {@code value}.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;))
+     * </pre>
+     *
+     * @param value Target value.
+     */
+    static Condition equalTo(byte[] value) {
```

Comment:
should we have similar overloaded methods for another operations?

Reply:
I can't imagine any example for such a case.

Reply:
For example I would select rows for particular values of bynary columns, Why I can not do it?
like SELECT * FROM T WHERE bin_col in ('0xAA','0xBB')
the same for NOT IN

Reply:
Added for in, not in